### PR TITLE
Network plots

### PR DESCRIPTION
**Pull request by clonker**

### CODE REVIEW

**Review by MaaikeG — COMMENTED** [review]
I have a few minor comments.

Also, I tried running the plot_network.py example locally, but I'm getting an error:
  File "(...)\Deeptime\deeptime\deeptime\plots\network.p
y", line 160, in cmap
    from matplotlib import colormaps
ImportError: cannot import name 'colormaps' from 'matplotlib' (C:\Users\Maaike\
iniconda3\envs\deeptime\lib\site-packages\matplotlib\__init__.py)

I have matplotlib version 3.4.3 installed.

**Comment by clonker** [comment]
thanks! does it work now?

File: deeptime/plots/network.py [hunk 1]
```diff
@@ -0,0 +1,523 @@
...
+        # set node colors
+        from matplotlib import colors
+        if isinstance(value, str):
+            state_colors = np.array([colors.to_rgba(value)] * self.n_nodes)
+        else:
+            state_colors = ensure_number_array(value)
+            if state_colors.ndim == 1:
+                state_colors = np.array([self.cmap(x) for x in state_colors])
+            elif state_colors.ndim in (3, 4):
+                pass  # ok
```
[the remark this answers is not in the document]
Comment (clonker) [reply deeptime/plots/network.py]: yeah actually this check was completely nonsensical

File: deeptime/plots/network.py [hunk 2]
```diff
@@ -0,0 +1,523 @@
...
+            raise ValueError("Invalid edge labels, should be 'weights', ndarray of strings or None.")
+
+    @property
+    def state_sizes(self) -> np.ndarray:
+        r""" Sizes of states. Can be left `None` which amounts to state sizes of 1. """
+        return self._state_sizes
+
+    @state_sizes.setter
+    def state_sizes(self, value: Optional[np.ndarray]):
+        self._state_sizes = value if value is not None else np.ones(self.n_nodes)
```
[the remark this answers is not in the document]
Comment (clonker) [reply deeptime/plots/network.py]: added

File: deeptime/plots/network.py [hunk 3]
```diff
@@ -0,0 +1,523 @@
...
+        element in the adjacency matrix """
+        if issparse(self.adjacency_matrix):
+            mat = self.adjacency_matrix.tocoo()
+            max_off_diag = -np.inf
+            for i, j, v in zip(mat.row, mat.col, mat.data):
+                if i != j:
+                    max_off_diag = max(max_off_diag, v)
+        else:
+            diag_mask = np.logical_not(np.eye(self.adjacency_matrix.shape[0], dtype=bool))
+            max_off_diag = np.max(self.adjacency_matrix[diag_mask & (self.adjacency_matrix > 0)])
```
[the remark this answers is not in the document]
Comment (clonker) [reply deeptime/plots/network.py]: i now check for largest off-diagonal value in absolute values

File: deeptime/plots/network.py [hunk 4]
```diff
@@ -0,0 +1,523 @@
...
+        The colormap for `state_color`s.
+    state_sizes : ndarray, optional, default=None
+        List of state sizes to plot, must be of length `n_states`. It effectively evaluates as
+
+        .. math::
+            \frac{\mathrm{state\_scale} \cdot (\min (d_x, d_y))^2}{2 n_\mathrm{nodes}}
+              \frac{\mathrm{state\_sizes}}{\| \mathrm{state\_sizes} \|_{\max}}
+
+        with :math:`\mathrm{state\_sizes}` interpreted as `1` in case of `None`. In particular this means that
+        the states scale their size with respect to the volume. I.e., `state_scale=[1, 2]` leads to the second
```
[the remark this answers is not in the document]
Comment (clonker) [reply deeptime/plots/network.py]: yes

File: deeptime/plots/network.py [hunk 5]
```diff
@@ -0,0 +1,523 @@
...
+
+            # add annotation
+            if self.state_labels is not None:
+                ax.text(self.pos[i][0], self.pos[i][1], self.state_labels[i], zorder=3, **textkwargs)
+
+        assert len(circles) == self.n_nodes, f"{len(circles)} != {self.n_nodes}"
+
+        # draw arrows
+        for i, j in zip(*np.triu_indices(self.n_nodes, k=1)):  # upper triangular indices with 0 <= i < j < n_nodes
+            if abs(self.adjacency_matrix[i, j]) > 0:
```
[the remark this answers is not in the document]
Comment (clonker) [reply deeptime/plots/network.py]: semantically equivalent but yours brings less computational overhead :slightly_smiling_face:

File: deeptime/plots/network.py [hunk 6]
```diff
@@ -0,0 +1,523 @@
...
+        if isinstance(value, str):
+            state_colors = np.array([colors.to_rgba(value)] * self.n_nodes)
+        else:
+            state_colors = ensure_number_array(value)
+            if state_colors.ndim == 1:
+                state_colors = np.array([self.cmap(x) for x in state_colors])
+            elif state_colors.ndim in (3, 4):
+                pass  # ok
+            else:
+                raise ValueError()
```
[the remark this answers is not in the document]
Comment (clonker) [reply deeptime/plots/network.py]: agreed

**Pull request merged**